Source/ASDisplayNode.mm

```diff
@@ -2741,7 +2741,9 @@ - (void)setHierarchyState:(ASHierarchyState)newState
   // Entered or exited range managed state.
   if ((newState & ASHierarchyStateRangeManaged) != (oldState & ASHierarchyStateRangeManaged)) {
     if (newState & ASHierarchyStateRangeManaged) {
-      [self enterInterfaceState:self.supernode.interfaceState];
+      if (self.supernode) {
+        [self enterInterfaceState:self.supernode->_pendingInterfaceState];
```

Comment:
In order to do this, we would need to grab the __instanceLock__ of the supernode, which shouldn't be done manually.

To correctly fix this case, we'll need to add an accessor method for - (ASInterfaceState)pendingInterfaceState which uses a MutexLocker and returns the _pendingInterfaceState.

This method can be declared in ASDisplayNode+FrameworkPrivate.h, so it can be used by various internal subclasses without being made public right away (we may eventually want to make this public, but not until there is a use case — I'm optimistic that there will never be a need to do this).